cfgov/core/utils.py

```diff
@@ -10,13 +10,26 @@
 from core.templatetags.svg_icon import svg_icon
 
 
-NON_GOV_LINKS = re.compile(
-    r"https?:\/\/(?:www\.)?(?![^\?]+\.gov)(?!(content\.)?localhost).*"
+LINK_PATTERN = re.compile(
+    r"^(?P<schema>https?)://(?P<domain>[^/:]+):?(?P<port>\d+)?(?P<path>/?.*)?$"
 )
+
 NON_CFPB_LINKS = re.compile(
     r"(https?:\/\/(?:www\.)?(?![^\?]*(cfpb|consumerfinance).gov)"
     r"(?!(content\.)?localhost).*)"
 )
+
+
+def should_interstitial(url: str) -> bool:
+    match = LINK_PATTERN.match(url)
+    if match.group("domain").endswith(".gov") and NON_CFPB_LINKS.match(url):
+        return False
+    return True
+
+
+NON_GOV_LINKS = re.compile(
+    r"https?:\/\/(?:www\.)?(?![^\?]+\.gov)(?!(content\.)?localhost).*"
+)
```

Comment:
To keep the blame history clean, can this regex move back to around line 13, so git sees it as not changes?